src/main/java/htsjdk/samtools/SamPairUtil.java

```diff
@@ -1,513 +1,513 @@
+        setMateInfo(rec1, rec2, addMateCigar);
+        setProperPairFlags(rec1, rec2, expectedOrientations);
+    }
+
+    public static void setProperPairFlags(final SAMRecord rec1, final SAMRecord rec2, final List<PairOrientation> expectedOrientations) {
+        final boolean properPair =  (!rec1.getReadUnmappedFlag() && !rec2.getReadUnmappedFlag())
+                ? isProperPair(rec1, rec2, expectedOrientations)
+                : false;
+        rec1.setProperPairFlag(properPair);
+        rec2.setProperPairFlag(properPair);
+    }
+
+    /**
+     * A class to iterate through SAMRecords and set mate information on the given records, and optionally
+     * set the mate cigar tag (true by default).
+     */
+    public static class SetMateInfoIterator extends PeekableIterator<SAMRecord> {
+
+        private final Queue<SAMRecord> records = new LinkedList<SAMRecord>();
+        private final boolean setMateCigar;
+        private final boolean ignoreMissingMates;
+        private long numMateCigarsAdded = 0;
+
+        /**
+         * By default, the mate cigar tag is set
+         * @param iterator the iterator to wrap
+         */
+        public SetMateInfoIterator(final Iterator<SAMRecord> iterator) {
+            this(iterator, true);
+        }
+
+        /**
+         * @param iterator the iterator to wrap
+         * @param setMateCigar true if we are to update/create the Mate CIGAR (MC) optional tag, false if we are to clear any mate cigar tag that is present.
+         */
+        public SetMateInfoIterator(final Iterator<SAMRecord> iterator, final boolean setMateCigar) {
+            this(iterator, setMateCigar, false);
+        }
+
+        /**
+         * @param iterator the iterator to wrap
+         * @param setMateCigar true if we are to update/create the Mate CIGAR (MC) optional tag, false if we are to clear any mate cigar tag that is present.
+         * @param ignoreMissingMates set this to true if we are to ignore missing mates, otherwise an exception will be thrown when a missing mate is encountered
+         */
+        public SetMateInfoIterator(final Iterator<SAMRecord> iterator, final boolean setMateCigar, final boolean ignoreMissingMates) {
+            super(iterator);
+            this.setMateCigar = setMateCigar;
+            this.ignoreMissingMates = ignoreMissingMates;
+        }
+
+        /**
+         * @return the current number of mate cigars added.  This could be more than the number of records returned.
+         */
+        public long getNumMateCigarsAdded() { return this.numMateCigarsAdded; }
+
+        @Override
+        public boolean hasNext() {
+            return (!records.isEmpty() || super.hasNext());
+        }
+
+        /**
+         * Populate this.records if necessary.
+         */
+        private void advance() {
+
+            // No need to advance if we have records remaining
+            if (!records.isEmpty()) return;
+
+            /*
```

Comment:
It's good to fix this -- it's not a valid javadoc so it shouldn't start with ** -- but somehow the entire file is showing up as changed, as a whitespace change.

Can you make another PR that doesn't show every line as being different? Also, multi-line comments usually have a `*` character at the start of each line. Or, you could use `//` for each line of a multi-line comment.